piplmesh/panels/horoscope/panel.py

```
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+
+import datetime
+
+from django.utils import translation
+from django.utils.translation import ugettext_lazy as _
+
+from piplmesh import panels
+
+from . import models, providers
+
+HOROSCOPE_DATE_TOO_OLD = 2
+
+class HoroscopePanel(panels.BasePanel):
+    def get_context(self, context):
+        user = context['user']
+
+        context.update({
+            'header': _("Today's horoscope"),
+        })
+
+        if not user.birthdate:
+            context.update({
+                'error_birthdate': True,
+            })
+            return context
+
+        try:
+            provider = providers.get_provider(translation.get_language())
+        except:
```

Don/t catch all exceptions!